runtime/commonMain/src/kotlinx/serialization/internal/NullableSerializer.kt

```diff
@@ -5,50 +5,64 @@
 package kotlinx.serialization.internal
 
 import kotlinx.serialization.*
-import kotlin.reflect.KClass
 
-fun <T : Any> makeNullable(actualSerializer: KSerializer<T>): KSerializer<T?> = NullableSerializer(actualSerializer)
 
-class NullableSerializer<T : Any>(public val actualSerializer: KSerializer<T>) : KSerializer<T?> {
-    private class SerialDescriptorForNullable(val original: SerialDescriptor): SerialDescriptor by original {
-        override val isNullable: Boolean
-            get() = true
-
-        override fun equals(other: Any?): Boolean {
-            if (this === other) return true
-            if (other !is SerialDescriptorForNullable) return false
-
-            if (original != other.original) return false
+/**
+ * Returns a nullable serializer for the given serializer of non-null type.
+ */
+public fun <T : Any> KSerializer<T>.nullable(): KSerializer<T?> {
```

Comment:
`KSerializer.list` and `KSerializer.map` are extension properties, so I propose to make `nullable` a property too for the sake of consistency

Reply:
I have questions for this design (and `Pair<Ser, Ser>.map` extension in particular).
Let's postpone it (or you could raise them at our DM), renamed for the sake of consistency